docs/DeveloperGuide.md

```diff
@@ -382,33 +369,40 @@ The timings that the persons are free at the specified date will be displayed.
+
+A successful execuction of the `findCommonTiming` command is described as follows:
 
-Given below is an example usage scenario and explanation on how the `findCommonTiming` command behaves at each step.
+1. `FindCommonTimingCommand` uses the predicate prepared during parsing to filter the list of persons in `Model`.
+2. Events occuring on a certain day are then retrieved using the `getEventsAtDate` function.
+3. The day will be represented as an array of 48 time slots, with each element of the array representing a 30-minute timeslot.
+4. TheThe `blockTimeSlots` function is used to ensure that respective timeslots are set as busy according to when events occur.
+5. Free time slots will be appended to a string that is due to be returned in the `CommandResult`.
+6. A `CommandResult` with the timeslots that the persons are free will be returned (timeslots are in intervals of 30 minutes). 
+These timeslots will then be displayed to the user.
 
-1. The user enters `findCommonTiming t/friends da/2022-03-04` to find the common timings that the persons who share the same tag are free. 
-The arguments `t/friends da/2022-03-04` are passed to the `findCommonTimingParser` through its `parse` method call.
+<img src="images/FindCommonTimingSequenceDiagram.png" />
 
-2. The user input `t/friends da/2022-03-04` will be checked to ensure that empty input is not given.
+### Design Considerations
 
-3. A new `IsTagInPersonPredicate` object is created and encapsulated by a new `FindCommonTiming` object.
+**Aspect: How FindCommonTiming executes**
+* **Alternative 1(current implementation)**: Retrieve all events' starting and ending times. Use an integer array to represent 24 hours in a day, and change the value of array elements at specific indices of the array according to event start time and duration of event. After all events are processed, print common free timings according to which elements of the array indicate a certain time slot to be free.
+  * Pros:
+    * The method is easier to grasp and has less edge cases that need to be handled.
+  * Cons:
+    * To print out free timings, a lot of effort is required to ensure timings to be printed out are bug-free.
 
-4. The `FindCommonTiming` object is returned to the `LogicManager`.
+* **Alternative 2**: Retrieve all events' starting and ending times. Have an array of objects belonging to a new class which has the attributes startTime and duration. The array would only have one element at first (00.00, 24). Split the elements in the array according to events' start Time and duration(for example if there is an event from 08.00-10.00, the array would contain (00.00, 8) and (10.00, 14). After all the events are processed, the array would contain the start time and duration of each block of free time available.
```

Comment:
Create a new `TimeSlot` class that has a start time and a duration attribute. Create an array containing 1 `TimeSlot` object with a start time of 00:00 and duration of 24 hours to represent the day. For every event that belongs to a person with the specified tag, get the start and end times of the event and split the `TimeSlot` object to "remove" the timeslots in which a person with the specified tag is not free. For instance, if the event is from 12:00-14:00, the `TimeSlot` object can be split into a `TimeSlot` with a start time of 00:00 and duration of 12 hours, and a `TimeSlot` with a start time of 14:00 and a duration of 10 hours. After every event is processed, return the array of `TimeSlot` objects.